pre_commit/commands/run.py

```diff
@@ -61,7 +61,11 @@ def get_filenames(args, include_expr, exclude_expr):
             lambda: get_changed_files(args.origin, args.source),
         )
     elif args.files:
-        getter = git.get_files_matching(lambda: args.files)
+        relative_files = [
+            os.path.relpath(os.path.abspath(filename), git.get_root())
+            for filename in args.files
+        ]
```

Comment:
I'd rather put this in main where all the code will benefit from it -- might make the testing more difficult (though there are some similar tests you can snitch from)

Alternately, you could split this functionality into a `normalize_arguments` function and write a test for that (so you don't have to bother with integration testing)